### PR TITLE
Feat monorepo refactoring

### LINKED ISSUE #55: Test Environment / Seperating Frontend and Backend
## User story
1. As a poweruser
2. I want / need seperated frontend and backend
3. So that it is modular and good for further developments

## Acceptance criteria
* test environment should be used 
* setup of the repository in monorepository
* seperating Frontend and Backend

## Definition of done (DoD)
* A Feature is Done when all its Acceptance criterias are fulfilled
* A Feauture is Done when the code has been peer reviewed from the dedicated team
* A Feature is Done when it has been tested
* A Feature is Done when its core functionality has been commented properly
* A Feature is Done when it can be fully integrated into the main branch without negative side effect

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

### PR DESCRIPTION
**Pull request by schlingling**
- added seperate packages for CLI, middleware and backend
- refactored code due to the separation in packaes (fixed filespathes and co)

@tawfeeka It would make sense, to merge this branch at first to main in order to avoid merge conflicts. Later on you can then merge your  PR

closes #55

### CODE REVIEW

**Review by tawfeeka — APPROVED** [review]
I wait until this branch is merged to merge cli
Nice job @schlingling

**Review by lukas-rotsching — COMMENTED** [review]
this should enhance really enhance the reusability especially of the backend. Good job.

**Pull request merged**
